Python/pywarpx/callbacks.py

```diff
@@ -10,36 +12,39 @@
+extra reference to the method's object is saved.
 
-The install can be done using a decorator, which has the prefix "callfrom". See example below.
+The install can be done using a decorator, which has the prefix "callfrom". See
+example below.
 
 Functions can be called at the following times:
- - afterinit <installafterinit>: immediately after the init is complete
- - beforeEsolve <installbeforeEsolve>: before the solve for E fields
- - poissonsolver <installpoissonsolver>: In place of the computePhi call but only in an electrostatic simulation
- - afterEsolve <installafterEsolve>: after the solve for E fields
- - beforedeposition <installbeforedeposition>: before the particle deposition (for charge and/or current)
- - afterdeposition <installafterdeposition>: after particle deposition (for charge and/or current)
- - beforestep <installbeforestep>: before the time step
- - afterstep <installafterstep>: after the time step
- - afterdiagnostics <installafterdiagnostics>: after diagnostic output
- - particlescraper <installparticlescraper>: just after the particle boundary conditions are applied
-                                             but before lost particles are processed
- - particleloader <installparticleloader>: at the time that the standard particle loader is called
- - particleinjection <installparticleinjection>: called when particle injection happens, after the position
-                                                  advance and before deposition is called, allowing a user defined
-                                                  particle distribution to be injected each time step
- - appliedfields <installappliedfields>: allows directly specifying any fields to be applied to the particles
-                                         during the advance
-
-To use a decorator, the syntax is as follows. This will install the function myplots to be called after each step.
+ - afterinit: immediately after the init is complete
+ - beforeEsolve: before the solve for E fields
+ - poissonsolver: In place of the computePhi call but only in an electrostatic
+                  simulation
+ - afterEsolve: after the solve for E fields
+ - beforedeposition: before the particle deposition (for charge and/or current)
+ - afterdeposition: after particle deposition (for charge and/or current)
+ - beforestep: before the time step
+ - afterstep: after the time step
+ - afterdiagnostics: after diagnostic output
+ - particlescraper: just after the particle boundary conditions are applied
+                    but before lost particles are processed
+ - particleloader: at the time that the standard particle loader is called
+ - particleinjection: called when particle injection happens, after the position
+                      advance and before deposition is called, allowing a user
+                      defined particle distribution to be injected each time step
+
+To use a decorator, the syntax is as follows. This will install the function
```

Comment:
How much do we care to support the decorator syntax going forward?

Reply:
That's a good point - I think we should generalize the callback decorator helpers to be more general and instead have one argument - the name of the callback. That way we can generalize them going forward.

cc @RemiLehe @dpgrote